pathfinder/pathfinding_service.py

```diff
@@ -203,7 +207,14 @@ def handle_token_network_created(self, event):
             self.create_token_network_for_address(token_network_address)
 
     def create_token_network_for_address(self, token_network_address: Address):
-        log.info(f'Following token network at {token_network_address}')
+        # check that code exists on the address
+        code_at_address = self.web3.eth.getCode(token_network_address)
```

Comment:
1) if you use `Web3` only in constructor, don't store it
2) it'd be better to inject this dependency - token networks can be checked before instantiating the `PathfindingService`
2a) if you want to check it in the constructor, create a helper function `contract_has_code() -> Bool`
3) what if the code disappears _after_ the check? ;)

Reply:
Good points.
1)+2) removed it from the pathfindin service, moved into the cli
3) that's a bummer, can a `TokenNetwork` destruct right now? Not sure if we want to handle this, in any case we show a nice backtrace 🚀